mne/encoding/model.py

```diff
@@ -0,0 +1,230 @@
+"""Base and high-level classes for fitting encoding models."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from ..externals.six import string_types
+
+
+class SampleMasker(object):
+    """Remove samples with a particular value before fitting.
+
+    Parameters
+    ----------
+    estimator : instance of sklearn-style estimator
+        The estimator to be called after samples are removed.
+    mask_val : float, int, nan | callable.
+        The value that will be removed from the dataset. If a callable,
+        it should return a boolean mask of length n_samples that takes
+        X an array of shape (n_samples, n_features) as input.
+    ixs : array, shape (n_samples,)
+        An optional array of indices for manually specifying which samples
+        to keep during training. Defaults to None.
+    ixs_pred : array, shape (n_samples,) | None
```

Comment:
Overall for these var names, it's better to explicit their meaning than their type